File: Homeworks/C#1/Operators-And-Expressions/CheckBit/CheckBit.cs
//Write a Boolean expression that returns if the bit 
//at position p (counting from 0, starting from the right) in given integer number n, has value of 1.
namespace CheckBit
{
    using System;
    class CheckBit
    {
        static void Main(string[] args)
        {
            Console.Write("Enter a number: ");
            int number = int.Parse(Console.ReadLine());
            Console.Write("Enter position to check bit: ");
            int position = int.Parse(Console.ReadLine());

            Console.WriteLine("Bit is one: " + BitIsOne(number, position));
        }

        private static bool BitIsOne(int number, int position)
        {
            if ((number & (1 << position)) == 0) return false;
            return true;
        }
    }
}

File: Homeworks/JS-FUNDAMENTALS/Operators-And-Expressions/checkBit.js
// Write a boolean expression for finding if the bit 3
// (counting from 0) of a given integer is 1 or 0.

var isOne = 8, isZero = 1;

console.log(getBitAt(isOne, 3));
console.log(getBitAt(isZero, 3));

function getBitAt(number, position){
    return (number >> position) & 1;
}